src/OrchardCore.Modules/OrchardCore.OpenId/Configuration/OpenIdValidationConfiguration.cs

```diff
@@ -208,19 +219,23 @@ public void Configure(string name, OAuthValidationOptions options)
             if (!string.IsNullOrEmpty(settings.Tenant) &&
                 !string.Equals(settings.Tenant, _shellSettings.Name, StringComparison.Ordinal))
             {
-                var context = _shellHost.GetOrCreateShellContext(_shellSettingsManager.GetSettings(settings.Tenant));
-                using (var scope = context.EnterServiceScope())
+                var shellSettings = _shellSettingsManager.GetSettings(settings.Tenant);
+
+                if (shellSettings.State == TenantState.Running)
```

Comment:
This is also problematic. Ideally, `_shellHost.EnterServiceScope` should take care of starting the server tenant if it's stopped at the time we reach this code path.

Reply:
This is what it does (see my other comment below), but:

I think that here, when you reference another tenant e.g it can't be `Uninitialized` or `Initializing` meaning it has not been created or setup. So, the only problem would be if it has been **disabled**.

When you get a shell settings, as long as it is not hold by an active shell context, it's state is not changed, but if it is already **disabled** `GetOrCreateShellContext()` will return a context wich is not released but with a null `ServiceProvider`. And you will never be able to create a service scope.

Note: If you already are in an active scope and the shell is disabled after, it is marked as released and removed from the shells dico. But you can still ends your scope with a non null provider. But on the next `GetOrCreateShellContext()` the shell will be re-created as described above.

Note: The `OC.Tenants` index page lists all shell contexts including disabled tenants.

In summary, maybe here the only problem is if the tenant is already disabled. And in this case you will never be able to create a service scope. Let me know what is better to do in this case.